doc/source/serve/core-apis.rst

```diff
@@ -1,128 +1,150 @@
-By default endpoints are serviced only by the backend provided to :mod:`create_endpoint <ray.serve.api.create_endpoint>`, but in some cases you may want to specify multiple backends for an endpoint, e.g., for A/B testing or incremental rollout.
-For information on how to do this, please see :ref:`serve-split-traffic`.
+  >> serve.list_deployments()
+  {'MyFirstDeployment': <ray.serve.api.ServeDeployment object at 0x7fe0c04c42b0>}
+
+  # Returns the same object as the original MyFirstDeployment object.
+  # This can be used to redeploy, get a handle, etc.
+  deployment = serve.get_deployment("MyFirstDeployment")
+
+Exposing a Deployment
+=====================
+
+By default, deployments are exposed over HTTP at ``http://localhost:8000/<deployment_name>``.
+The HTTP path that the deployment is available at can be changed using the ``route_prefix`` option.
+All requests to ``/{route_prefix}`` and any subpaths will be routed to the deployment (using a longest-prefix match for overlapping route prefixes).
+
+Here's an example:
 
 .. code-block:: python
 
-  serve.create_endpoint("simple_endpoint", backend="simple_backend", route="/simple", methods=["GET"])
+  @serve.deployment(name="http_deployment", route_prefix="/api")
+  class HTTPDeployment:
+    def __call__(self, request):
+        return "Hello world!"
 
-After creating the endpoint, it is now exposed by the HTTP server and handles requests using the specified backend.
+After creating the endpoint, it is now exposed by the HTTP server and handles requests using the specified class.
 We can query the model to verify that it's working.
 
 .. code-block:: python
 
   import requests
-  print(requests.get("http://127.0.0.1:8000/simple").text)
+  print(requests.get("http://127.0.0.1:8000/api").text)
 
 We can also query the endpoint using the :mod:`ServeHandle <ray.serve.handle.RayServeHandle>` interface.
 
 .. code-block:: python
 
-  handle = serve.get_handle("simple_endpoint")
+  # To get a handle from the same script, use the Deployment object directly:
+  handle = HTTPDeployment.get_handle()
+
+  # To get a handle from a different script, reference it by name:
+  handle = serve.get_deployment("http_deployment").get_handle()
+
   print(ray.get(handle.remote()))
 
-To view all of the existing endpoints that have created, use :mod:`serve.list_endpoints <ray.serve.api.list_endpoints>`.
+Updating a Deployment
+=====================
+
+Often you want to be able to update your code or configuration options for a deployment over time.
+Deployments can be updated simply by updating the code or configuration options and calling ``deploy()`` again.
 
 .. code-block:: python
 
-  >>> serve.list_endpoints()
-  {'simple_endpoint': {'route': '/simple', 'methods': ['GET'], 'traffic': {}}}
+  @serve.deployment(name="my_deployment", num_replicas=1)
```

Comment:
add comment this is the SAME file over time

Reply:
done